src/02-common-principles.md

```diff
@@ -363,6 +363,102 @@ In particular, if a BIDS dataset contains a `derivatives/` sub-directory,
 the contents of that directory may be a heterogeneous mix of BIDS Derivatives
 datasets and non-compliant derivatives.
 
+## BIDS URI: Pointing to files within and outside of BIDS datasets
+
+Throughout the BIDS specification there a meta data fields that contain a
+pointer to a file.
+For example `IntendedFor`, `SpatialReference`, and `AssociatedEmptyRoom`.
+Prior to BIDS version 1.6.1, such pointers were required to be "relative paths".
+However the specification failed to consistently  describe relative to *what*
+these pointers should be specified, and the exact syntax to be used.
+Yet even with perfectly well and consistently defined "relative paths", this
+method of pointing to files becomes impractical when referring to files that
+are outside of a particular file tree (for example, on the Internet), or when
+the location of the data that is being pointed to changes.
+
+For these reasons, starting with BIDS version 1.6.1, using "relative paths" for
+all fields pointing to a separate file is [DEPRECATED][].
+
+Instead, the following [URI][uniform-resource-indicator] scheme SHOULD be
+used to point to files within and outside of BIDS datasets, as elaborated below.
+
+`bids:<dataset-name>:/absolute/path/within/dataset`
+
+Here, `bids:` is the URI scheme, and `<dataset-name>:/absolute/path/within/dataset`
+is the path to the file, consisting of the name of a dataset, a colon, and the absolute
+path within that dataset that MUST start with a forward slash `/`.
+
+The location of a dataset with the name `<dataset-name>` MUST be specified in the
+[`dataset_description.json` file](./03-modality-agnostic-files.md#dataset-description)
+under the `DatasetLinks` field, which is an [object][] of [strings][], as shown in
+the examples below.
+Note that the `<dataset-name>`: `"local"` is a reserved value that may only be
+used to refer to the current dataset.
+
+### Refer to a file within a dataset
+
+To link for example to a derivative file that is stored in a nested `derivatives/`
+directory within a dataset:
+`bids:deriv1:/sub-01/anat/sub-01_desc-preproc_T1w.nii.gz`
+
+```json
+{
+    "DatasetLinks": {
+        "deriv1": "file://derivatives/derivative1"
+    }
+
+}
+```
+
+Note that this is the same as specifying the following pointer:
+`bids:local:/derivatives/derivative1/sub-01/anat/sub-01_desc-preproc_T1w.nii.gz`
+Because the reserved `<dataset-name>`: `"local"` always points to the
+root of the current dataset.
+The `DatasetLinks` metadata does not have to be specified if `local` is the
+only `<dataset-name>` that occurs throughout the dataset.
+
+### Refer to a file outside of a dataset but on the same host
```

Comment:
the examples should also show how to point to a file that is on the same Windows computer, e.g. on a external USB drive that is mounted as `F:\` (and note the slash going the other way here)

Reply:
Good point, my understanding would be that the `/` refers to the "root of the file system". That is, the "shallowest" point you can navigate to, if you keep typing `cd ..` for as long as it makes a difference. On unix systems, `/` coincides with that "shallowest" point in the file system.

My assumption is that a Windows file system also has such a "shallowest" point. And if that assumption is true and we agree on using `file:///` to represent that, specifying the path to an external USB-mounted drive at `F:\` should become trivial (unless Windows file systems work completely differently to unix file systems, please correct me in that case).

Note that as is already common practice in BIDS, we don't allow backslashes for paths of any kind --> I thought that Windows can deal with forward slashes for paths, again please correct me if I am wrong here.

Reply:
Typically you'd assume a "virtual" root on Windows systems, which would be `/`. `C:\Windows` would have the URI `file:///C/Windows`

I'd vote against support for the backslash character – all directory paths should be normalized use forward slashes.

> I thought that Windows can deal with forward slashes for paths, again please correct me if I am wrong here.

It can't, but it's trivial to convert between the two. I believe, however, that BIDS should support one, and only one, way to specify paths, to make parsing easier.

Reply:
this discussion was continued and then concluded in https://github.com/bids-standard/bids-specification/pull/820#discussion_r648369579

What Richard suggested above is correct, except that my tests have shown that the colon after `C` is needed: `file:///C:/...`